dbfit-java/core/src/main/java/dbfit/api/PreparedDbCommand.java

```diff
@@ -0,0 +1,18 @@
+package dbfit.api;
+
+import java.sql.SQLException;
+
+/**
+ * Represents a prepared database statement with ability to bind input parameters and
+ * to register and access output ones including return values.
+ */
+public interface PreparedDbCommand extends DbCommand, DbQuery {
```

Comment:
> So, perhaps we'd only have to know if we're creating an INSERT statement or not, but more specifically whether we, potentially, need to get at any generated keys and this would only be where we're building an INSERT statement for the Insert fixture I think.

Right now `getGeneratedKeys` is accessed via `DbParameterAccessor` - the database statement member in it (which  in turn is being set via `bindTo`).

In general - I think we should be aiming to easy extension and maintenance in terms of:
  1. New environment implementations (and let's say - some of these may be proprietary). Things should be simple but also stable for environment authors. (Adding more methods to `DBEnvironment` interface is in general a pain for environment implementations: it may break them /it's forcing them to be implemented/). So in such situations "fat" interfaces might be a sane choice (e.g. JDBC API is like that).
  2. New test systems (in addition to Fit), maybe new fixtures, etc. The API used by the fixtures should be thin and simple (yet, adding more methods here is not a pain /removing ones would be/). By providing a bridging layer between `DBEnvironment` and the fixtures - we can avoid exposing excessive details to fixtures.

These are just general comments. It might be worthwhile to see if we can come up with something better about auto-generated keys access.